src/utils/validation/validateSync/applyFormRules.js

```diff
@@ -21,6 +22,9 @@ function reduceRules(rules) {
 export default function applyFormRules(rules) {
   return (resolverArgs) => {
     const rulesList = listOf(
+      /** @todo This is weird */
+      addWhen(R.path(['fieldPath', 'name'], rules), isset, reduceRules),
```

Comment:
This entire logic can be eliminated by filtering out empty rules key paths on `getFieldRules` level.
The whole call signature here is very confusing, and should be removed.